runtime/test/reference-test.js

```diff
@@ -358,4 +357,163 @@ describe('references', function() {
+      manifest: `
+        schema Result
+          Text value
+        
+        particle ExtractReference in 'extractReference.js'
+          in Foo {[Reference<Result>] result} referenceIn
+          out [Result] rawOut
+          
+        recipe
+          create 'input:1' as handle0
+          create 'output:1' as handle1
+          ExtractReference
+            referenceIn <- handle0
+            rawOut -> handle1
+        `,
+      'extractReference.js': `
+        defineParticle(({Particle}) => {
+          return class Dereferencer extends Particle {
+            setHandles(handles) {
+              this.output = handles.get('rawOut');
+            }
+
+            async onHandleSync(handle, model) {
+            }
+
+            async onHandleUpdate(handle, update) {
+              if (handle.name == 'referenceIn') {
+                for (const result of update.data.result) {
+                  await result.dereference();
+                  this.output.store(result.entity);
+                }
+              }
+            }
+
+            onHandleDesync(handle) {
+            }
+          }
+        });
+      `
+    });
+
+    let pecFactory = function(id) {
+      let channel = new MessageChannel();
+      new ParticleExecutionContext(channel.port1, `${id}:inner`, loader);
+      return channel.port2;
+    };
+    let arc = new Arc({id: 'test:0', pecFactory, loader});
+
+    let manifest = await Manifest.load('manifest', loader);
+    let recipe = manifest.recipes[0];    
+    assert.isTrue(recipe.normalize());
+    assert.isTrue(recipe.isResolved());
+    await arc.instantiate(recipe);
+
+    const inMemoryEngine = arc._storageProviderFactory._storageInstances['in-memory'];
+    const baseStoreType = Type.newEntity(manifest.schemas.Result);
+    const backingStore = await inMemoryEngine.baseStorageFor(baseStoreType, inMemoryEngine.baseStorageKey(baseStoreType));
+    await backingStore.store({id: 'id:1', rawData: {value: 'what a result!'}}, ['totes a key']);
+    await backingStore.store({id: 'id:2', rawData: {value: 'what another result!'}}, ['totes a key']);
+
+    const refStore = arc._stores[1];
+    assert.equal(refStore._type.entitySchema.name, 'Foo');
+    await refStore.set({id: 'id:a', rawData: {result: [{id: 'id:1', storageKey: backingStore.storageKey}, {id: 'id:2', storageKey: backingStore.storageKey}]}});
+
+    await arc.idle;
+    const outputStore = arc._stores[0];
+    assert.equal(outputStore._type.getContainedType().entitySchema.name, 'Result');
+    const values = await outputStore.toList();
+    assert.equal(values.length, 2);
+    assert.equal(values[0].rawData.value, 'what a result!');
+    assert.equal(values[1].rawData.value, 'what another result!');
+  });
+
+  it('can construct collections of references in schemas', async () => {
+    let loader = new StubLoader({
+      manifest: `
+        schema Result
+          Text value
+        
+        particle ExtractReference in 'extractReference.js'
```

Comment:
ExtractReference is a C&P from the above test, maybe update the name to something like "CreateReference" ?

Reply:
Done.